src/fmu/tools/rms/generate_petro_jobs_for_field_update.py

```diff
@@ -0,0 +1,506 @@
+):
+    nvar_new = len(new_var_names)
+    if nvar_new == 1:
+        # No correlation matrix
+        return new_var_names, []
+
+    # Order the var_names_to_keep list to be in same sequence
+    # as orig_var_names for those variables that are common with the var_names_to_keep
+    # Example: the keep list is: ["A", "C",  "B"]
+    # and the original var name list
+    # is ["A", "D", "F", "B", "C"]
+    # The sorted keep list should be: ["A", "B", "C"]
+
+    sorted_new_var_names = sort_new_var_names(
+        orig_var_names, facies_name, new_var_names
+    )
+    new_corr_matrix = []
+    index_in_orig_var = []
+    for i, var_name in enumerate(orig_var_names):
+        name = set_new_var_name(facies_name, var_name)
+        if name in new_var_names:
+            index_in_orig_var.append(i)
+
+    for row in range(nvar_new):
+        row_list = []
+        for col in range(row):
+            row_list.append(
+                orig_corr_matrix[index_in_orig_var[row]][index_in_orig_var[col]]
+            )
+        new_corr_matrix.append(row_list)
+
+    return sorted_new_var_names, new_corr_matrix
+
+
+def sort_new_var_names(original_variable_names, facies_name, new_variable_names):
+    sorted_keep_list = []
+    for varname in original_variable_names:
+        name = set_new_var_name(facies_name, varname)
+        if name in new_variable_names:
+            sorted_keep_list.append(name)
+    return sorted_keep_list
+
+
+def get_original_job_settings(owner_string_list, job_type, job_name):
+    original_job = roxar.jobs.Job.get_job(owner_string_list, job_type, job_name)
+    return original_job.get_arguments(skip_defaults=False)
+
+
+def create_copy_of_job(
+    owner_string_list, job_type, original_job_arguments, new_job_name
+):
+    new_job = roxar.jobs.Job.create(owner_string_list, job_type, new_job_name)
+    new_job.set_arguments(original_job_arguments)
+    return new_job
+
+
+def get_zone_names_per_facies(used_petro_per_zone_per_facies_dict):
+    zone_names_with_facies_dict = {}
+    for zone_name, facies_dict in used_petro_per_zone_per_facies_dict.items():
+        for facies_name, _ in facies_dict.items():
+            if facies_name not in zone_names_with_facies_dict:
+                zone_names_with_facies_dict[facies_name] = []
+            zone_names_with_facies_dict[facies_name].append(zone_name)
+    return zone_names_with_facies_dict
+
+
+def get_used_petro_names(used_petro_per_zone_per_facies_dict):
+    all_petro_var_list = []
+    for _, petro_per_facies_dict in used_petro_per_zone_per_facies_dict.items():
+        for _, petro_list in petro_per_facies_dict.items():
+            for petro_name in petro_list:
+                if petro_name not in all_petro_var_list:
+                    all_petro_var_list.append(petro_name)
+    return all_petro_var_list
+
+
+def set_new_var_name(facies_name, petro_name):
+    return facies_name + "_" + petro_name
+
+
+def check_consistency(
```

Comment:
Can you break this function into two (as it is long) and also add a docstring what it will do?

Reply:
Fixed